torch/fx/symbolic_trace.py

```diff
@@ -432,7 +432,14 @@ def forward(*args, **kwargs):
 
         self.submodule_paths = None
 
-        return self.graph
+        # Since the tracer might be packaged along with a GraphModule that it helped generate, and that
+        # graphs are explicilty not saved, set self.graph to None so that it doesn't get packaged
+        # along with the tracer.
+        graph = self.graph
```

Comment:
This isn't really actionable for this PR, but we should clarify what the expected tracer semantics are (and give guidance to custom tracer writers). In particular, we should make it clear that `trace()` should not permanently mutate tracer state, so that we can guarantee that a tracer can always be used to reconstruct a graphmodule even across a serialization boundary.

Looks like right now we informally enforce this (we blank out some of the scratch state like `submodule_paths` and reinitialize some other state like `self.root` and `self.graph`) but it would be good to have a documented expectation. @jamesr66a what do you think?